app/src/views/features/sessions/SessionViewer/DraftSessionViewer.tsx

```diff
@@ -109,8 +163,25 @@ const DraftSessionViewer: React.FC = () => {
           name: "Mock Session Recording",
         })
       );
+
       dispatch(sessionRecordingActions.setEvents(mockSession.events));
       setIsLoading(false);
+    } else if (queryParams.has("saved")) {
+      StorageService()
+        .getRecord(APP_CONSTANTS.DRAFT_SESSIONS)
+        .then((sessions) => {
+          const session = sessions[tabId];
+          const sessionEvents = decompressEvents(session.events);
+          dispatch(
+            sessionRecordingActions.setSessionRecordingMetadata({
+              sessionAttributes: session.metadata,
+              name: generateDraftSessionTitle(session.metadata?.url),
+            })
+          );
+          filterOutLargeNetworkResponses(sessionEvents);
+          dispatch(sessionRecordingActions.setEvents(sessionEvents));
+          setIsLoading(false);
+        });
```

Comment:
Can we move these into a function so that this `useEffect` becomes more readable? Also the else part here should be moved